services/src/main/java/org/keycloak/services/resources/SessionCodeChecks.java

```diff
@@ -222,9 +224,9 @@ public boolean initialVerify() {
         setClientToEvent(client);
         session.getContext().setClient(client);
 
-        if (!client.isEnabled()) {
+        if (!client.isEnabled() && clientCode != null) {
```

Comment:
One note: Is it possible to make sure that `SessionCodeChecks` is "effectively unchanged" ?

The thing is, that `SessionCodeChecks` is used also during authentication and I fear of the possible security issues. I am thinking about something like:
- add the protected method like `checkClientDisabled` in the `SessionCodeChecks` class. This protected method will do exactly same thing as today in the `SessionCodeChecks` class. Something like:
```
protected boolean checkClientDisabled() {
    return !client.isEnabled;
}
- In the `LogoutSessionCodeChecks`, this protected method can be overriden and check changed to something like:
```
protected boolean checkClientEnabled() {
    return !client.isEnabled && clientCode != null;
}
```
Will this work?

Reply:
I understand. A strategy model to ensure that according to the operation we will carry out the necessary verification. Implemented as per your guidance.